pkg/register/authentication.go

```diff
@@ -26,37 +27,37 @@ func IsAllowFor(resolverClient ResolverClient, issuer *Issuer, issuerDoc *Regist
 		return false, err
 	}
 	if !delegationProof.Revoked {
-		err := ValidateDelegation(resolverClient, subjectDoc.ID, delegationProof)
+		err := ValidateDelegation(ctx, resolverClient, subjectDoc.ID, delegationProof)
 		return err == nil, err
 	}
 	return false, fmt.Errorf("delegation proof revoked")
 }
 
-func checkAllowOnDocOrController(resolverClient ResolverClient, issuer *Issuer, subjectID string, includeAuth bool) error {
-	issuerDoc, err := resolverClient.GetDocument(issuer.Did)
+func checkAllowOnDocOrController(ctx context.Context, resolverClient ResolverClient, issuer *Issuer, subjectID string, includeAuth bool) error {
+	issuerDoc, err := resolverClient.GetDocument(ctx, issuer.Did)
 	if err != nil {
 		return err
 	}
 
-	subjectDoc, err := resolverClient.GetDocument(subjectID)
+	subjectDoc, err := resolverClient.GetDocument(ctx, subjectID)
 	if err != nil {
 		return err
 	}
 
-	if allowed, err := IsAllowFor(resolverClient, issuer, issuerDoc, subjectDoc, includeAuth); allowed {
+	if allowed, err := IsAllowFor(ctx, resolverClient, issuer, issuerDoc, subjectDoc, includeAuth); allowed {
 		return nil
 	} else if IsResolverError(err) {
```

Comment:
context errors mustn't be ignored here. Could use a helper like in ebus (` IsContextError`) to check for here (and below for other usage of `IsResolverError`)

Reply:
[90e52eb](https://github.com/Iotic-Labs/iotics-identity-go/pull/46/commits/90e52eb275d81d88961688ed35415e343cc869b2)